spec/unit/provider/group/dscl_spec.rb

```diff
@@ -117,20 +117,27 @@
     before do
       @node = Chef::Node.new
       @provider = Chef::Provider::Group::Dscl.new(@node, @new_resource)
-      allow(@provider).to receive(:safe_dscl).and_return("\naj      500\n")
     end
 
-    it "should run safe_dscl with list /Groups gid" do
-      expect(@provider).to receive(:safe_dscl).with(*"list /Groups gid".split(" "))
+    it "should run safe_dscl with search /Groups gid" do
+      expect(@provider).to receive(:safe_dscl).with(*"search /Groups gid 500".split(" ")).and_return('')
       @provider.gid_used?(500)
     end
 
     it "should return true for a used gid number" do
+      allow(@provider).to receive(:safe_dscl).and_return(<<-eos
+        someprogram		somethingElse:gid = (
+            500
+        )
+        eos
+      )
       expect(@provider.gid_used?(500)).to be_truthy
     end
 
     it "should return false for an unused gid number" do
+      allow(@provider).to receive(:safe_dscl).and_return('')
       expect(@provider.gid_used?(501)).to be_falsey
```

Comment:
for example here you want to make sure 50, 00, 5 don't match just because 500 is in use. so having the same mocked output is useful.